lib/src/layer/polyline_layer.dart

```diff
@@ -127,95 +115,149 @@ class PolylinePainter extends CustomPainter {
   final bool saveLayers;
 
   final FlutterMapState map;
+  final double zoom;
+  final double rotation;
 
-  PolylinePainter(this.polylineOpt, this.saveLayers, this.map);
+  PolylinePainter(this.polylines, this.saveLayers, this.map)
+      : zoom = map.zoom,
+        rotation = map.rotation;
+
+  int get hash {
+    _hash ??= Object.hashAll(polylines);
+    return _hash!;
+  }
+
+  int? _hash;
+
+  List<Offset> getOffsets(List<LatLng> points) {
+    return points.map((pos) => getOffset(pos)).toList();
```

Comment:
```suggestion
    return List.generate(points.length, (index) {
      return getOffset(points[index]);
    }, growable: false);
```

Use a list generator again for performance reasons. However we will not embed getOffset since it is used in other parts of the code here

Reply:
Done